### PR TITLE
[language ☑, swarm ⍰, types ⍰][janitorial, easy] Unwrap removal + combinator patterns 

### PR DESCRIPTION
**Pull request by huitseeker**
- cleans up a couple of unnecessary unwraps w/ try syntax
- cleans up Result / Option / iter combinator patterns

### CODE REVIEW

File: types/src/transaction.rs [hunk 1]
```diff
@@ -1061,10 +1061,7 @@ impl TransactionListWithProof {
     }
 
     fn display_option_version(version: Option<Version>) -> String {
-        match version {
-            Some(v) => format!("{}", v),
-            None => String::from("absent"),
-        }
+        version.map_or_else(|| String::from("absent"), |v| format!("{}", v))
```
Comment (bmwill) [inline types/src/transaction.rs]: I don't really know if this is really more readable.

**Comment by vgao1996** [comment]
The changes in language looks good to me. Thanks for the cleanup!

File: language/compiler/src/unit_tests/testutils.rs [hunk 2]
```diff
@@ -35,7 +35,7 @@ fn compile_script_string_impl(
     code: &str,
     deps: Vec<CompiledModule>,
 ) -> Result<(CompiledScript, Vec<VMStatus>)> {
-    let parsed_program = parse_program(code).unwrap();
```
Comment (zamsden) [inline language/compiler/src/unit_tests/testutils.rs]: I'd actually recommend leaving the unwrap() calls here - in unit tests, it is easier to deal with a panic at the source than a Result() that gets passed back up the stack.

File: libra_swarm/src/swarm.rs [hunk 3]
```diff
@@ -301,7 +301,7 @@ impl LibraSwarm {
             .with_faucet_keypair(faucet_account_keypair)
             .with_role(role)
             .with_upstream_config_dir(upstream_config_dir.clone());
-        let config = config_builder.build().unwrap();
+        let config = config_builder.build()?;
```
Comment (zamsden) [inline libra_swarm/src/swarm.rs]: That one is definitely a bug.

File: types/src/validator_verifier.rs [hunk 4]
```diff
@@ -192,10 +192,11 @@ impl<PublicKey: VerifyingKey> ValidatorVerifier<PublicKey> {
     where
         T: Into<PublicKey::SignatureMaterial> + Clone,
     {
-        for author in aggregated_signature.keys() {
-            if self.author_to_public_keys.get(&author) == None {
-                return Err(VerifyError::UnknownAuthor);
-            }
+        if aggregated_signature
```
Comment (davidiw) [inline types/src/validator_verifier.rs]: What is the motivation for this type of conversion? From a readability perspective, both of them seem equally accessible.

**Comment by huitseeker** [comment]
@davidiw re: more diffs coming: I don't know. When I find a wart, I try to fix similar occurrences.

File: types/src/validator_verifier.rs (hunk 4 above)
Comment (huitseeker) [reply types/src/validator_verifier.rs]: Thanks for the pushback, uploading a more readable version.

**Comment by zamsden** [comment]
@bors-libra r+

Looks good to me, thanks for the cleanup!

**Pull request closed without merging**